x/migration/types/morse_claimable_account.go

```diff
@@ -1,7 +1,74 @@
 package types
 
+import (
+	"context"
+	"math/big"
+	"time"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/pokt-network/poktroll/app/pocket"
+)
+
 // IsClaimed returns true if the MorseClaimableAccount has been claimed;
 // i.e. ShannonDestAddress is not empty OR the ClaimedAtHeight is greater than 0.
 func (m *MorseClaimableAccount) IsClaimed() bool {
 	return m.ShannonDestAddress != "" || m.ClaimedAtHeight > 0
 }
+
+// IsUnbonding indicates that the MorseClaimableAccount began unbonding on Morse
+// but its unbonding peroid has NOT yet elapsed.
+func (m *MorseClaimableAccount) IsUnbonding() bool {
+	// DEV_NOTE: The UnstakingTime field is a time.Time type, which has a zero value of "0001-01-01T00:00:00Z" when printed as an ISO8601 string.
+	// See: https://pkg.go.dev/time#Time.IsZero
+	return !m.UnstakingTime.IsZero()
+}
+
+// HasUnbonded indicates that the MorseClaimableAccount began unbonding on Morse
+// and the unbonding period has elapsed.
```

Comment:
Can you #PUC when this will happen?

E.g. Supplier was claimed > 21 days after it started unbonding.